File: Move_Zeros/moving_zeros.cs
class Solution {
    public void moveZeroes(int[] nums) {
	int start = 0;
	int end = nums.Length - 1;
	int x = 0, i = 0;
	int k = end;
	int[] newArr = new int[end + 1];
	if(start < end)
	{
		while(start <= end)
		{
			if(nums[start] == 0)
			{
				newArr[k] = 0;
				k--;
				start++;
			}
			else
			{
				newArr[x] = nums[start];
				start++;
				x++;
			}
		}
		while(i <= end)
		{
			nums[i] = newArr[i];
			i++;
		}
	}
    }
}

File: Move_Zeros/moving_zeros.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i, j = 0, k;
        int end = nums.length - 1;
        int count = 0;
        
        for(i = 0; i <= end; i++)
        {
            if(nums[i] != 0)
            {
                nums[j] = nums[i];
                j++;
                count++;
            }
        }
        for(k = count; k <= end; k++)
        {
            nums[k] = 0;
        }
    }
}